src/value/borrowed.rs

```diff
@@ -123,12 +124,32 @@ pub struct BorrowedStruct<'val> {
     no_text_fields: Vec<(BorrowedSymbolToken<'val>, BorrowedElement<'val>)>,
 }
 
-impl<'val> BorrowedStruct<'val> {
-    // TODO remove constructor and instead add from_iter function to HashMap and Vec from iterator
-    fn new(
-        text_fields: HashMap<String, Vec<(BorrowedSymbolToken<'val>, BorrowedElement<'val>)>>,
-        no_text_fields: Vec<(BorrowedSymbolToken<'val>, BorrowedElement<'val>)>,
-    ) -> Self {
+impl<'val, K, V> FromIterator<(K, V)> for BorrowedStruct<'val>
+where
+    K: Into<BorrowedSymbolToken<'val>>,
+    V: Into<BorrowedElement<'val>>,
+{
+    /// Returns a borrowed struct from the given iterator of field names/values.
+    fn from_iter<I: IntoIterator<Item = (K, V)>>(iter: I) -> Self {
+        let mut text_fields: HashMap<String, Vec<(BorrowedSymbolToken, BorrowedElement)>> =
```

Comment:
Food for thought: we could optimize for the common case by using [`SmallVec`](https://doc.servo.org/smallvec/struct.SmallVec.html) instead of `Vec`.

Reply:
Added #154 for this.